### ISSUE TITLE
#expect(throws:) does not work with @MainActor

### ISSUE THREAD

**Issue by DagAgren** [swiftlang/swift-testing#866]
### Description

`#expect(throws:)` with an async block seems to get confused when run in a `@MainActor` function:

```
import Testing

struct Test {
	@Test func works() async throws {
		await #expect(throws: TestError.self) {
			try await Task.sleep(nanoseconds: 1)
			throw TestError()
		}
	}

	@Test @MainActor func doesNotWork() async throws {
		await #expect(throws: TestError.self) {
			try await Task.sleep(nanoseconds: 1)
			throw TestError()
		}
	}
}

struct TestError: Error {}
```

### Expected behavior

Both of these should compile and run fine.

### Actual behavior

The second function causes mysterious errors inside the code generated by the macro.

### Steps to reproduce

Compile the code above.

### swift-testing version/commit hash

Whatever ships with Xcode 16.0

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0
Darwin Mac 24.0.0 Darwin Kernel Version 24.0.0: Tue Sep 24 23:38:45 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T8122 arm64

**Comment by grynspan** [comment]
I am not able to reproduce any issue here using the 6.0.2 toolchain. I had to rename your test suite from `Test` as it conflicts with symbols exposed by Swift Testing.

There was an actor isolation fix implemented in 6.0.1. Please try updating your copy of Xcode to something newer than 16.0; if that doesn't help, we can reopen.

**Comment by DagAgren** [comment]
Can confirm that it does work on later Xcode! Thanks for pointing me at that.
